corehq/apps/es/management/commands/elastic_sync_multiplexed.py

```diff
@@ -0,0 +1,222 @@
+import math
+from datetime import datetime
+
+from django.core.management.base import BaseCommand, CommandError
+
+from corehq.apps.es.client import (
+    BulkActionItem,
+    ElasticMultiplexAdapter,
+    Tombstone,
+    get_client,
+)
+from corehq.apps.es.registry import get_registry
+from corehq.apps.es.transient_util import _DOC_ADAPTERS_BY_INDEX
+from corehq.apps.hqcase.management.commands.reindex_es_native import (
+    check_task_progress,
+)
+
+
+class ESSyncUtil:
```

Comment:
Most of the functionality in this class has been moved to the adapter but I would still prefer to keep this class in the same file as it eases out testing the management command.

Reply:
If the methods in the class are only meant to be used by this command, moving them directly to the command, or making these private util methods rather than a class, would fit more into our standard code patterns, and signal to others not to use them. It would be a pretty strong anti pattern for future users to import a utility method from this command (similar to moving check progress, like you did below), so marking them explicitly private would be nice, and would continue to allow easy testing.

Reply:
I personally would not prefer a lot of private methods scattered over file which are supporting one thing. 
If it is used somewhere else in the future it would be a good candidate to be moved to the utils file rather than keeping it in the management command as we did with check_progress.
I have added this as a comment https://github.com/dimagi/commcare-hq/pull/32370/commits/56703f4d9bfb4c9065843c8ab052fe54db768275 in util class